source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Charges/Handlers/ChargePriceEventHandlerTests.cs

```diff
@@ -135,7 +142,55 @@ public class ChargePriceEventHandlerTests
 
             // Assert
             chargePriceRejectionService.Verify(
-                x => x.SaveRejections(It.IsAny<ChargePriceOperationsRejectedEvent>()), Times.Once);
+                x => x.Publish(It.IsAny<PriceRejectedEvent>()), Times.Once);
+        }
+
+        [Theory]
+        [InlineAutoMoqData]
+        public async Task GiveHandleAsync_WhenValidationFails_ThenValidationErrorsAreLogged(
```

Comment:
```suggestion
        public async Task HandleAsync_WhenValidationFails_ThenValidationErrorsAreLogged(
```

OR

```suggestion
        public async Task GivenHandleAsync_WhenValidationFails_ThenValidationErrorsAreLogged(
```

I tend to like the first suggestion most.

Reply:
done